drivers/stepper/step_dir/step_dir_stepper_common.h

```diff
@@ -45,7 +45,7 @@ struct step_dir_stepper_common_config {
 	{                                                                                          \
 		.step_pin = GPIO_DT_SPEC_GET(node_id, step_gpios),                                 \
 		.dir_pin = GPIO_DT_SPEC_GET(node_id, dir_gpios),                                   \
-		.dual_edge = DT_PROP(node_id, dual_edge_step),                                     \
+		.dual_edge = DT_PROP_OR(node_id, dual_edge_step, false),                           \
```

Comment:
This commit is not required, i suppose :)

Reply:
It's needed otherwise the dual-edge-step property needs to be part of either the driver dts or the stepper-controller dts binding. You might not want to do this if the IC does not support this setting. But if the property is not present in dts compilation with just `DT_PROP` will fail.

This way only drivers that have that setting can declare it in their own dts, otherwise its just false.